### PR TITLE
Add capability to store information about the plasma state during iterations

### PR DESCRIPTION
**Pull request by unoebauer**
This PR adds the capability to store and later access information about the plasma state during the different Monte Carlo iterations. Specifically, the inner boundary temperature, the dilution factor, the radiation temperature and the electron densities are stored.

- [x] add storage for plasma information during iterations
- [x] add storage to HDF5 dump
- [x] add appropriate tests
- [x] add docstrings
- [x] update documentation
- [x] update reference data

### CODE REVIEW

**Comment by unoebauer** [comment]
Just extended test_simulation.py to include a check for the newly stored information of the plasma state during the iterations. Naturally, I'll need to update the reference data.

**Comment by unoebauer** [comment]
@wkerzendorf - next week we should discuss how to best proceed about updating the reference data since multiple PRs require this step.

File: docs/running/access_iterations.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+**********************************************
+Access information about individual iterations
+**********************************************
+
+Currently we store information about the plasma state and the inner boundary
+for each iteration. This is saved in the simulation object and can be accesses
```
Comment (chvogl) [inline docs/running/access_iterations.rst]: accessed

**Comment by unoebauer** [comment]
@wkerzendorf - we could start merging this one. How should we proceed? First merge the respective PR on the ref-data repo?

**Pull request merged**